lib/internal/util/parse_args/parse_args.js

```diff
@@ -118,12 +128,22 @@ function checkOptionUsage(config, token) {
  * @param {string|undefined} optionValue - value from user args
  * @param {object} options - option configs, from parseArgs({ options })
  * @param {object} values - option values returned in `values` by parseArgs
+ * @param {boolean} allowNegative - allow negative optinons if true
  */
-function storeOption(longOption, optionValue, options, values) {
+function storeOption(longOption, optionValue, options, values, allowNegative) {
   if (longOption === '__proto__') {
     return; // No. Just no.
   }
 
+  if (allowNegative && StringPrototypeStartsWith(longOption, 'no-')) {
+    // Boolean option negation: --no-foo
+    const longOptionWithoutPrefixNo = StringPrototypeSlice(longOption, 3);
+    if (optionsGetOwn(options, longOptionWithoutPrefixNo, 'type') !== 'string') {
```

Comment:
The test in `checkOptionUsage` is against `'boolean'` so I think probably clearer and more consistent to test against `'boolean'` here?

```suggestion
    if (optionsGetOwn(options, longOptionWithoutPrefixNo, 'type') === 'boolean') {
```

Reply:
> The test in `checkOptionUsage` is against `'boolean'` so I think probably clearer and more consistent to test against `'boolean'` here?

here I have considered if options come from CLI, just like `node index.js --no-foo`, in this case,  `(optionsGetOwn(options, longOptionWithoutPrefixNo, 'type')` will be `undefined`, so it also should be added when using `if (... === 'boolean')`, which is better for consistent  testing.

Reply:
Ah, so `strict:false` and `allowNegation` means `--no-foo` returns `foo:false`. Ok, I like that. 👍 

This is very edge case, but what about `strict:false` and `--no-foo=x`? I think to be consistent with general behaviour of `--x=y` in `strict:false` should probably return `no-foo:x`?

Reply:
In which case, could perhaps change
```
  if (allowNegative && StringPrototypeStartsWith(longOption, 'no-')) {
```

to

```
  if (allowNegative && StringPrototypeStartsWith(longOption, 'no-') && optionValue === undefined) {
```